services/efs-guardian/tests/unit/test_efs_removal_policy_task.py

```diff
@@ -0,0 +1,175 @@
+    EfsManager,
+    NodeID,
+    ProjectID,
+)
+
+pytest_simcore_core_services_selection = ["postgres", "redis"]
+pytest_simcore_ops_services_selection = [
+    "adminer",
+]
+
+
+@pytest.fixture
+def app_environment(
+    monkeypatch: pytest.MonkeyPatch,
+    app_environment: EnvVarsDict,
+    postgres_env_vars_dict: EnvVarsDict,
+    with_disabled_rabbitmq_and_rpc: None,
+    wait_for_postgres_ready_and_db_migrated: None,
+    with_disabled_redis_and_background_tasks: None,
+):
+    # set environs
+    monkeypatch.delenv("EFS_GUARDIAN_POSTGRES", raising=False)
+
+    return setenvs_from_dict(
+        monkeypatch,
+        {
+            **app_environment,
+            **postgres_env_vars_dict,
+            "POSTGRES_CLIENT_NAME": "efs-guardian-service-pg-client",
+            "EFS_REMOVAL_POLICY_TASK_AGE_LIMIT_TIMEDELTA": "3600",
+        },
+    )
+
+
+@pytest.fixture
+async def user_in_db(
+    app: FastAPI,
+    user: dict[str, Any],
+    user_id: UserID,
+) -> AsyncIterator[dict[str, Any]]:
+    """
+    injects a user in db
+    """
+    assert user_id == user["id"]
+    async with insert_and_get_row_lifespan(  # pylint:disable=contextmanager-generator-missing-cleanup
+        app.state.engine,
+        table=users,
+        values=user,
+        pk_col=users.c.id,
+        pk_value=user["id"],
+    ) as row:
+        yield row
+
+
+@pytest.fixture
+async def project_in_db(
+    app: FastAPI,
+    user_in_db: dict,
+    project_data: dict[str, Any],
+    project_id: ProjectID,
+) -> AsyncIterator[dict[str, Any]]:
+    """
+    injects a project in db
+    """
+    assert f"{project_id}" == project_data["uuid"]
+    async with insert_and_get_row_lifespan(  # pylint:disable=contextmanager-generator-missing-cleanup
+        app.state.engine,
+        table=projects,
+        values=project_data,
+        pk_col=projects.c.uuid,
+        pk_value=project_data["uuid"],
+    ) as row:
+        yield row
+
+
+@patch("simcore_service_efs_guardian.services.background_tasks.get_redis_lock_client")
+@patch("simcore_service_efs_guardian.services.background_tasks.lock_project")
+async def test_efs_removal_policy_task(
+    mock_lock_project: MagicMock,
+    mock_get_redis_lock_client: MagicMock,
+    faker: Faker,
+    app: FastAPI,
+    efs_cleanup: None,
+    project_id: ProjectID,
+    node_id: NodeID,
+    project_in_db: dict,
+):
+    # 1. Nothing should happen
+    await removal_policy_task(app)
+    assert not mock_lock_project.called
+
+    # 2. Lets create some project with data
+    aws_efs_settings: AwsEfsSettings = app.state.settings.EFS_GUARDIAN_AWS_EFS_SETTINGS
+    _storage_directory_name = faker.word()
+    _dir_path = (
+        aws_efs_settings.EFS_MOUNTED_PATH
+        / aws_efs_settings.EFS_PROJECT_SPECIFIC_DATA_DIRECTORY
+        / f"{project_id}"
+        / f"{node_id}"
+        / f"{_storage_directory_name}"
+    )
+    efs_manager: EfsManager = app.state.efs_manager
+
+    with patch(
```

Comment:
suggestion:
```python
with mocker.patch( ... )
```

Reply:
It doesn't work out of the box as with unittest.mock module, what is the benefit of your approach?